### PR TITLE
Add retry to ApplicationDeployer delete

### PR DESCRIPTION
**Pull request by BrennanConroy**
@Tratcher @JunTaoLuo @victorhurdugaci

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/ApplicationDeployer.cs [hunk 1]
```diff
@@ -90,15 +90,21 @@ protected void CleanPublishedOutput()
             }
             else
             {
-                try
```
Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/ApplicationDeployer.cs]: Extract this into a helper function?

File: src/Microsoft.AspNetCore.Server.IntegrationTesting/Common/RetryHelper.cs [hunk 2]
```diff
@@ -72,5 +72,24 @@ public class RetryHelper
...
+        public static void RetryOperation(
+            Action retryBlock,
+            Action<Exception> exceptionBlock,
+            int retryCount = 3)
+        {
+            for (var retry = 0; retry < retryCount; ++retry)
+            {
+                try
+                {
+                    retryBlock();
```
Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting/Common/RetryHelper.cs]: Might want to add a slight delay between retries

**Comment by JunTaoLuo** [comment]
:shipit:

File: src/Microsoft.AspNetCore.Server.IntegrationTesting/Common/RetryHelper.cs [hunk 3]
```diff
@@ -72,5 +72,27 @@ public class RetryHelper
...
+            int retryDelayMilliseconds = 0)
+        {
+            for (var retry = 0; retry < retryCount; ++retry)
+            {
+                try
+                {
+                    retryBlock();
+                    break;
+                }
+                catch (Exception exception)
```
Comment (victorhurdugaci) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting/Common/RetryHelper.cs]: Don't handle the exception here. Let the caller decide if they want or not to handle it and make `retryBlock` return a boolean that tells the caller if it should retry or not

Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting/Common/RetryHelper.cs]: Actually why not just throw the exception when the retry limit has been reached?

Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting/Common/RetryHelper.cs]: Which is what the RetryRequest does.

Comment (victorhurdugaci) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting/Common/RetryHelper.cs]: I'm talking about something like this: https://github.com/aspnet/Logging/blob/e9e62d27d3719fe0abbb94ca93f5c2bc2d80fffd/test/Microsoft.Extensions.Logging.AzureWebAppDiagnostics.Test/RetryHelper.cs

**Pull request merged**